sklearn/linear_model/_coordinate_descent.py

```diff
@@ -186,9 +191,13 @@ def _alpha_grid(
         n_samples = sample_weight.sum()
     else:
         n_samples = X.shape[0]
+
+    if positive:
+        Xyw = np.maximum(0, Xyw)
+
     # Compute np.max(np.sqrt(np.sum(Xyw**2, axis=1))). We switch sqrt and max to avoid
-    # many computations of sqrt. This, however, needs an additional np.abs.
-    alpha_max = np.sqrt(np.max(np.abs(np.sum(Xyw**2, axis=1)))) / (n_samples * l1_ratio)
+    # many computations of sqrt.
+    alpha_max = np.sqrt(np.max(np.sum(Xyw**2, axis=1))) / (n_samples * l1_ratio)
```

Comment:
```suggestion
    if not positive:
        # Compute np.max(np.sqrt(np.sum(Xyw**2, axis=1))). We switch sqrt and max to avoid
        # many computations of sqrt.
        alpha_max = np.sqrt(np.max(np.sum(Xyw**2, axis=1))) / (n_samples * l1_ratio)
    else:
        # We may safely assume Xyw.shape[1] == 1, MultiTask estimators do not support positive constraints.
        alpha_max = max(0, np.max(Xyw)) / (n_samples * l1_ratio)
```